framework/src/application/application.ts

```diff
@@ -82,29 +83,85 @@ const registerProcessHooks = app => {
 			},
 			'System error: unhandledRejection',
 		);
-		app.shutdown(1, err.message);
+		// eslint-disable-next-line @typescript-eslint/no-floating-promises
+		app.shutdown(1, (err as Error).message);
 	});
 
-	process.once('SIGTERM', () => app.shutdown(1));
+	// eslint-disable-next-line @typescript-eslint/no-misused-promises
+	process.once('SIGTERM', async () => app.shutdown(1));
 
-	process.once('SIGINT', () => app.shutdown(1));
+	// eslint-disable-next-line @typescript-eslint/no-misused-promises
+	process.once('SIGINT', async () => app.shutdown(1));
 
-	process.once('exit', (error, code) => app.shutdown(code, error));
+	// eslint-disable-next-line
+	process.once('exit' as any, (code: number) => app.shutdown(code));
 };
 
-class Application {
-	constructor(genesisBlock, config = {}) {
+interface ApplicationConfig {
+	label: string;
+	version: string;
+	protocolVersion: string;
+	networkId: string;
+	lastCommitId: string;
+	buildVersion: string;
+	ipc: {
+		enabled: boolean;
+	};
+	tempPath: string;
+	readonly forging: {
+		readonly waitThreshold: number;
+		readonly delegates: DelegateConfig[];
+		readonly force?: boolean;
+		readonly defaultPassword?: string;
+	};
+	readonly rebuildUpToRound: string;
+	readonly network: NetworkConfig;
+	constants: {
+		[key: string]: {} | string | number | undefined;
```

Comment:
Why would the constant value be `undefined`?

Reply:
This `undefined` is to specify the key might not exist.
when we do `constants['some-key']`, it will be required to check if the constant exist or not